app/models/service_template.rb

```diff
@@ -75,6 +78,10 @@ class ServiceTemplate < ApplicationRecord
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
 
+  def self.public_service_templates
+    ServiceTemplate.all - ServiceTemplate.joins(:reserved_rec).where(:reserves=> {:reserved => {:internal => true}.to_yaml}).to_a
```

Comment:
1. This should return a scope if possible
2. That where clause is wrong, because there could be other things in reserves if we need to add them.

Reply:
Also, please add a spec

Reply:
@Fryguy what is the way to write the `where` clause? We tested but cannot make `reserved` field deserialized in this query.

Reply:
Can we query like `ServiceTemplate.includes(:reserved_rec).select { |s| s.internal == true }`?

Reply:
@Fryguy Changed back to scope based on above query.